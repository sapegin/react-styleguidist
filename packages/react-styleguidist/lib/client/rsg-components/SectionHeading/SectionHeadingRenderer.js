import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import Heading from 'rsg-components/Heading';
import Styled from 'rsg-components/Styled';

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function SectionHeadingRenderer(_ref) {
  const classes = _ref.classes;

      
const children = _ref.children;

      
const toolbar = _ref.toolbar;

      
const id = _ref.id;

      
const href = _ref.href;

      
const depth = _ref.depth;

      
const deprecated = _ref.deprecated;
  const headingLevel = Math.min(6, depth);
  const sectionNameClasses = cx(classes.sectionName, _defineProperty({}, classes.isDeprecated, deprecated));
  return React.createElement("div", {
    className: classes.wrapper
  }, React.createElement(Heading, {
    level: headingLevel,
    id
  }, React.createElement("a", {
    href,
    className: sectionNameClasses
  }, children)), React.createElement("div", {
    className: classes.toolbar
  }, toolbar));
}

const styles = function styles(_ref2) {
  const color = _ref2.color;

      
const space = _ref2.space;
  return {
    wrapper: {
      display: 'flex',
      flexDirection: 'row',
      alignItems: 'center',
      marginBottom: space[1]
    },
    toolbar: {
      marginLeft: 'auto'
    },
    sectionName: {
      '&:hover, &:active': {
        isolate: false,
        textDecoration: 'underline',
        cursor: 'pointer'
      }
    },
    isDeprecated: {
      color: color.light,
      '&, &:hover': {
        textDecoration: 'line-through'
      }
    }
  };
};

SectionHeadingRenderer.propTypes = {
  classes: PropTypes.object.isRequired,
  children: PropTypes.node,
  toolbar: PropTypes.node,
  id: PropTypes.string.isRequired,
  href: PropTypes.string.isRequired,
  depth: PropTypes.number.isRequired,
  deprecated: PropTypes.bool
};
export default Styled(styles)(SectionHeadingRenderer);