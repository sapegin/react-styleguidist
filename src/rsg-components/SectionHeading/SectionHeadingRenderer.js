import React, { Component } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import Heading from 'rsg-components/Heading';

import './index.scss';

export default class SectionHeadingRenderer extends Component {

  static propTypes = {
    children: PropTypes.node,
    toolbar: PropTypes.node,
    id: PropTypes.string.isRequired,
    href: PropTypes.string.isRequired,
    depth: PropTypes.number.isRequired,
    deprecated: PropTypes.bool,
    type: PropTypes.string,
  };

  render() {
    const headingLevel = Math.min(6, this.props.depth);
    return (
      <div className={cx({
        'section-heading__wrapper': true,
        'section-heading--deprecated': this.props.deprecated,
      })}>
        <Heading level={headingLevel} id={this.props.id} type={this.props.type} section>
          <a href={this.props.href} className="section-heading__sectionName">
            {this.props.children}
          </a>
        </Heading>
        <div className="section-heading__toolbar">{this.props.toolbar}</div>
      </div>
    );
  }
}
