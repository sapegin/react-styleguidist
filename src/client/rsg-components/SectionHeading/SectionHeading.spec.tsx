import React from 'react';
import { mount } from 'enzyme';
import SectionHeadingRenderer from './SectionHeadingRenderer';

describe('SectionHeading', () => {
	const FakeToolbar = () => <div>Fake toolbar</div>;

	test('render a section heading', () => {
		const actual = mount(
			<SectionHeadingRenderer id="section" href="/section" depth={2} toolbar={<FakeToolbar />}>
				A Section
			</SectionHeadingRenderer>
		);

		expect(actual.find('h2')).toMatchSnapshot();
	});

	test('render a deprecated section heading', () => {
		const actual = mount(
			<SectionHeadingRenderer
				id="section"
				href="/section"
				depth={2}
				toolbar={<FakeToolbar />}
				deprecated
			>
				A Section
			</SectionHeadingRenderer>
		);

		expect(actual.find('h2')).toMatchSnapshot();
	});

	test('prevent the heading level from exceeding the maximum allowed by the Heading component', () => {
		const actual = mount(
			<SectionHeadingRenderer id="section" href="/section" depth={7} toolbar={<FakeToolbar />}>
				A Section
			</SectionHeadingRenderer>
		);

		expect(actual.find('h6')).toHaveLength(1);
	});
});
