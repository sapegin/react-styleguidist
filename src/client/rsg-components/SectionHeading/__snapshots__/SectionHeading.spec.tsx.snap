// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionHeading render a deprecated section heading 1`] = `
<h2
  className="rsg--heading-5 rsg--heading2-7"
  id="section"
>
  <a
    className="rsg--sectionName-2 rsg--isDeprecated-3"
    href="/section"
  >
    A Section
  </a>
</h2>
`;

exports[`SectionHeading render a section heading 1`] = `
<h2
  className="rsg--heading-5 rsg--heading2-7"
  id="section"
>
  <a
    className="rsg--sectionName-2"
    href="/section"
  >
    A Section
  </a>
</h2>
`;
